contracts/invariant.ral

```diff
@@ -579,7 +579,15 @@ Contract Invariant(
 
         return matchingPools        
     }
-    pub fn getAllPoolKeys() -> ByteVec {
-        return wrappedGetAllPoolKeys()
+
+    pub fn getPoolKeys(size: U256, offset: U256) -> (ByteVec, U256) {
+        let mut slice = #
+        let offsetEnd = offset + size
+        let max = if (offsetEnd > poolKeyCount) poolKeyCount else offsetEnd
```

Comment:
the variable's name is a bit confusing, you are computing a min, but the variable is called `max`, I get the intent after thoroughly reading though, my suggestion - `sliceEnd`